GamebookEngine/Data/GameDatabase.swift

```diff
@@ -256,6 +256,25 @@ class GameDatabase {
         completion(nil)
     }
 
+    func searchPages(for game: Game, terms: String, completion: @escaping ([Page]) -> Void) {
+        fetchAllPages(for: game) { pages in
+            guard let pages = pages else {
+                completion([])
+                return
```

Comment:
You don't need this `return`, because `completion` is a replacement for return, basically.

Reply:
If I leave out the return statement, I got the following error: ```'guard' body must not fall through, consider using a 'return' or 'throw' to exit the scope```

Reply:
Oh, huh! My bad. What happens if you remove the `completion([])`? Might be fine to just have the `return` alone in that case.

Reply:
This is interesting, might be appropriate here:

https://medium.com/swiftcraft/swift-alley-defer-statements-b0286bdad496

Reply:
If I remove completion([]), it compiles and the search works so far. The defer statement looks interesting, and it is the first time I have seen it.

Reply:
Looking into it more, I dunno if `defer` is appropriate here.

Excluding the `completion` in the guard makes me worried that we're setting up a situation where the calling method is waiting for completion and never receives it.

According to [this page](https://appventure.me/posts/2016-03-29-three-tips-for-clean-swift-code.html), this would be okay:

```swift
fetchAllPages(for: game) { pages in
    guard let pages = pages else { return completion([]) }
    ...
}
```

Reply:
> ```guard let pages = pages else { return completion([]) }```

This works.

Reply:
Nice! 👍